From 258d547d636399a8ac3204c1cf29d8036a7bb50d Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Mon, 11 Aug 2014 16:19:22 -0400
Subject: [idle] intel_idle: close avn_cstates array with correct marker

Message-id: <20140811161922.48098.4679.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 87635
O-Subject: [PATCH RHEL7.1 BZ1048050 3/3] intel_idle: close avn_cstates array with correct marker
Bugzilla: 1048050
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1048050

Description:
------------
Close avn_cstates array with correct marker to avoid overflow
in function intel_idle_cpu_init().

[rjw: The problem was introduced when commit 22e580d07f65 was merged
 on top of eba682a5aeb6 (intel_idle: shrink states tables).]

Fixes: 22e580d07f65 (intel_idle: Fixed C6 state on Avoton/Rangeley processors)
Signed-off-by: Jiang Liu <jiang.liu@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=88390996c95b879ba365888199b45ace3f5ca80b

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index d168122..10b86fa 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -428,7 +428,7 @@ static struct cpuidle_state atom_cstates[] = {
  {
   .enter = NULL }
 };
-static struct cpuidle_state avn_cstates[] __initdata = {
+static struct cpuidle_state avn_cstates[] = {
  {
   .name = "C1-AVN",
   .desc = "MWAIT 0x00",
@@ -443,6 +443,8 @@ static struct cpuidle_state avn_cstates[] __initdata = {
   .exit_latency = 15,
   .target_residency = 45,
   .enter = &intel_idle },
+ {
+  .enter = NULL }
 };
 
 /**
-- 
1.7.1